application/F3DConfigFileTools.cxx

```diff
@@ -217,26 +219,46 @@ F3DConfigFileTools::ReadConfigFiles(const std::string& userConfig)
         {
           // Add each options config entry into an option dict
           F3DOptionsTools::OptionsDict entry;
+          F3DOptionsTools::OptionsDict imperativeEntry;
           for (const auto& item : optionsBlock.items())
           {
+            // Individual item can be imperative, store in the right dict
+            std::string key = item.key();
+            F3DOptionsTools::OptionsDict* localEntry = &entry;
+            if (!key.empty() && key[0] == '!')
+            {
+              localEntry = &imperativeEntry;
+              key = key.substr(1);
+            }
```

Comment:
Pointers are so much C 😛 
```suggestion
            std::reference_wrapper<F3DOptionsTools::OptionsDict> localEntry = std::ref(entry);
            if (!key.empty() && key[0] == '!')
            {
              localEntry = std::ref(imperativeEntry);
              key = key.substr(1);
            }
```

Reply:
when you dont know how to write something, just write it wrong and ask for a review :p